### ISSUE TITLE
Mine and Grains not refreshing, scheduling broken in salt-minion

### ISSUE THREAD

**Issue by nmadhok** [saltstack/salt#31609]
### Description of Issue/Question
Seeing error along with traceback in the minion logs. Minion doesn't seem to update mine or grains and scheduling is broken:
```
root@nitin-saltminion:~# salt-minion -l error
[ERROR   ] Unhandled exception running event.fire
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/utils/schedule.py", line 703, in handle_func
    ret['retcode'] = self.functions.pack['__context__']['retcode']
KeyError: 'retcode'
[ERROR   ] Unhandled exception running mine.update
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/utils/schedule.py", line 703, in handle_func
    ret['retcode'] = self.functions.pack['__context__']['retcode']
KeyError: 'retcode'
[ERROR   ] Unhandled exception running event.fire
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/utils/schedule.py", line 703, in handle_func
    ret['retcode'] = self.functions.pack['__context__']['retcode']
KeyError: 'retcode'
```

### Setup
Install latest salt-minion package available from EPEL. In the minion config file, add the following:
```
grains_refresh_every: 1
```

### Steps to Reproduce Issue
Start minion in error log mode:
```
root@nitin-saltminion:~# salt-minion -l error
[ERROR   ] Unhandled exception running event.fire
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/utils/schedule.py", line 703, in handle_func
    ret['retcode'] = self.functions.pack['__context__']['retcode']
KeyError: 'retcode'
[ERROR   ] Unhandled exception running mine.update
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/utils/schedule.py", line 703, in handle_func
    ret['retcode'] = self.functions.pack['__context__']['retcode']
KeyError: 'retcode'
[ERROR   ] Unhandled exception running event.fire
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/utils/schedule.py", line 703, in handle_func
    ret['retcode'] = self.functions.pack['__context__']['retcode']
KeyError: 'retcode'
```

### Versions Report
```
Salt Version:
           Salt: 2015.8.7
 
Dependency Versions:
         Jinja2: unknown
       M2Crypto: 0.20.2
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
         Python: 2.6.6 (r266:84292, Jul 23 2015, 05:13:40)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: 3.2.2
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed
 
System Versions:
           dist: oracle 6.7 
        machine: x86_64
        release: 3.8.13-118.2.5.el6uek.x86_64
         system: Oracle Linux Server 6.7 
```

**Comment by nmadhok** [comment]
This issue seems to have been discussed in an existing issue (https://github.com/saltstack/salt/issues/31106) and a fix for this issue is a part of the PR #31189 which has been merged. Closing this issue in favor of the fix.

**Comment by rukender** [comment]
Hi nmadhok ,

could you please help me get the solution for this issue. After I upgrade salt minions to 2016.11.0 version 
File "/usr/lib/python2.6/site-packages/salt/utils/schedule.py", line 703, in handle_func
    ret['retcode'] = self.functions.pack['__context__']['retcode']
KeyError: 'retcode'
2017-01-03 06:51:16,056 [salt.utils.schedule                                  ][ERROR   ][26458] Unhandled exception running mine.update
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/utils/schedule.py", line 703, in handle_func
    ret['retcode'] = self.functions.pack['__context__']['retcode']
KeyError: 'retcode'
2017-01-03 07:51:16,071 [salt.utils.schedule                                  ][ERROR   ][7282] Unhandled exception running mine.update
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/utils/schedule.py", line 703, in handle_func
    ret['retcode'] = self.functions.pack['__context__']['retcode']
KeyError: 'retcode'
2017-01-03 08:03:27,440 [salt.loader                                          ][ERROR   ][11220] Failed to import beacons win_pulsar, this is due most likely to a syntax error:
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/loader.py", line 1135, in _load_module
    ), fn_, fpath, desc)
  File "/var/cache/salt/minion/extmods/beacons/win_pulsar.py", line 420
     events_list.append({k: event_dict[k] for k in ('EntryType', 'Accesses', 'TimeGenerated', 'Object Name', 'Hash')})
